Feature support go wasm

Muhammad-Ayman opened the pull request:
## What type of PR is this? (check all applicable)

- [x] ✨ Feature

## Description

Added initial support for running Go code in the browser using Yaegi (Go interpreter) compiled to WebAssembly.
## Related Tickets & Documents

see https://github.com/live-codes/livecodes/issues/796


## Added tests?
- [x] 🙋 no, because I need help

## Added to documentations?
- [x] 📜 README.md



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Go (Wasm) language added with an in-browser Yaegi-based runtime and interactive run/readiness API for demos.
  - Go (Wasm) Starter template added and selectable in templates and starter menus.

- Documentation
  - New Go (Wasm) docs page, expanded Go docs, updated translations, and languages badge bumped.
  - Vendor license entries added for Yaegi/yaegi-wasm.

- Chores
  - Build and vendor configuration updated to include Go (Wasm).
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

hatemhosny commented:
Thank you very much @Muhammad-Ayman 
This is really impressive work

> I’ve also published a new version **`yaegi-wasm@1.0.2`** on the CDN. This release now supports **stdin**, completes the **starter template**, and includes the **language info section**.

Nice 👍 

> Regarding the origin, I tried using `*` as you suggested, but unfortunately **SonarQubeCloud is refusing it**.

I have marked that as safe in SonarQube

> As for regenerating the WASM file, I should mention that I **manually compiled the Yaegi interpreter to WASM**, so at the moment I don’t know how to automate that process.

You did it yourself? wow! 
Good job. It works very well.

For regenerating the wasm for new versions, can you please document the steps so that we can do that later for new versions? This can be in the README of your repo for example.

> * _compilation errors reporting_
> * _running more Go playground examples (especially tests)_
> 
> I don’t have a clear idea yet on how to approach these, and I’m currently a bit busy to explore them. But for now, here we go with the updates above .

No worries. These are not high priority for now. We can explore that later.

I have added a few edits and added [docs](https://deploy-preview-866--livecodes.netlify.app/docs/languages/go-wasm).
Would you please review that and tell me what you think?

If you are fine with the changes, I think this PR would be ready to merge.
Thanks a lot. That was such a helpful contribution.

Muhammad-Ayman commented:
> Thank you very much @Muhammad-Ayman This is really impressive work
> 
> > I’ve also published a new version **`yaegi-wasm@1.0.2`** on the CDN. This release now supports **stdin**, completes the **starter template**, and includes the **language info section**.
> 
> Nice 👍
> 
> > Regarding the origin, I tried using `*` as you suggested, but unfortunately **SonarQubeCloud is refusing it**.
> 
> I have marked that as safe in SonarQube
> 
> > As for regenerating the WASM file, I should mention that I **manually compiled the Yaegi interpreter to WASM**, so at the moment I don’t know how to automate that process.
> 
> You did it yourself? wow! Good job. It works very well.
> 
> For regenerating the wasm for new versions, can you please document the steps so that we can do that later for new versions? This can be in the README of your repo for example.
> 

Thanks a lot for the kind words! 🙏  
I’ve just added the documentation for regenerating the WASM build in the [go-wasm.mdx](https://github.com/Muhammad-Ayman/livecodes/blob/Feature_Support_Go_wasm/docs/docs/languages/go-wasm.mdx) as you suggested. Please let me know if anything needs more detail or clarification.  

> > * _compilation errors reporting_
> > * _running more Go playground examples (especially tests)_
> > 
> > I don’t have a clear idea yet on how to approach these, and I’m currently a bit busy to explore them. But for now, here we go with the updates above .
> 
> No worries. These are not high priority for now. We can explore that later.
> 
> I have added a few edits and added [docs](https://deploy-preview-866--livecodes.netlify.app/docs/languages/go-wasm). Would you please review that and tell me what you think?
> 
> If you are fine with the changes, I think this PR would be ready to merge. Thanks a lot. That was such a helpful contribution.

The edits and docs you added look great to me. I think this PR is ready to go. 🚀  

Thanks again for the guidance and support!

Muhammad-Ayman commented:
> I would appreciate if you add a license so that LiveCodes can use your package. 

Thank you for the suggestion!  

I’ve added a `Apache-2.0 license` to the project. I also updated the `package.json` to reflect the same license.
